Update code example from supabase.mdx (Astro & Supabase Guide) (#8799)
withastro/docs: src/content/docs/en/guides/backend/supabase.mdx

- `POST /api/auth/signin`: to sign in a user.
- `GET /api/auth/signout`: to sign out a user.

> Create these endpoints in the `src/pages/api/auth` directory of your project. Make sure to specify `export const prerender = false` to render the endpoints on demand. Your project should now include these new files:

Reviewer: ```suggestion
Create these endpoints in the `src/pages/api/auth` directory of your project. If you are using `hybrid` rendering mode, you must specify `export const prerender = false` at the top of each file to render these endpoints on demand. Your project should now include these new files:
```
This is only for `hybrid` mode, not `server` mode, so we we should make that clear!

`register.ts` creates a new user in Supabase. It accepts a `POST` request with the an email and password. It then uses the Supabase SDK to create a new user.

```ts title="src/pages/api/auth/register.ts"

> export const prerender = false;

Reviewer: ```suggestion
// With `output: 'hybrid'` configured:
// export const prerender = false;
```
Maybe something like this for these lines? This line is not necessarily correct for every project.

`signin.ts` signs in a user. It accepts a `POST` request with the an email and password. It then uses the Supabase SDK to sign in the user.

```ts title="src/pages/api/auth/signin.ts"

> export const prerender = false;

Reviewer: ```suggestion
// With `output: 'hybrid'` configured:
// export const prerender = false;
```

`signout.ts` signs out a user. It accepts a `GET` request and removes the user's access and refresh tokens.

```ts title="src/pages/api/auth/signout.ts"

> export const prerender = false;

Reviewer: ```suggestion
// With `output: 'hybrid'` configured:
// export const prerender = false;
```
